### PR TITLE
Gen4Fallback bug fix

### PR DESCRIPTION
**Pull request by systay**
## Description
A minor planner fix

When using the `Gen4fallback` planner, we were not protecting the AST from changes from the Gen4 planner, which lead to the V3 planner getting an already updated AST and failing

### CODE REVIEW

**Review by GuptaManan100 — APPROVED** [review]

File: go/vt/vtgate/planbuilder/abstract/join.go [hunk 1]
```diff
@@ -58,8 +58,8 @@ func (j *Join) PushPredicate(expr sqlparser.Expr, semTable *semantics.SemTable)
 			j.LeftJoin = false
 			return j.RHS.PushPredicate(expr, semTable)
 		}
-		// TODO - we should do this on the vtgate level once we have a Filter primitive
-		return vterrors.Errorf(vtrpcpb.Code_UNIMPLEMENTED, "unsupported: cross-shard left join and where clause")
+		semTable.ProjectionErr = vterrors.Errorf(vtrpcpb.Code_UNIMPLEMENTED, "unsupported: cross-shard left join and where clause")
```
Comment (GuptaManan100) [inline go/vt/vtgate/planbuilder/abstract/join.go:61]: We should rename this from `ProjectionErr` to something else, since it now contains errors that are more general than just projection errors

**Review by harshit-gangal — COMMENTED** [review]
looks good, need a test for it :)

File: go/vt/vtgate/planbuilder/abstract/join.go (hunk 1 above)
Comment (systay) [reply go/vt/vtgate/planbuilder/abstract/join.go:61]: :+1:

File: go/vt/vtgate/planbuilder/abstract/join.go [hunk 2]
```diff
@@ -58,8 +58,7 @@ func (j *Join) PushPredicate(expr sqlparser.Expr, semTable *semantics.SemTable)
 			j.LeftJoin = false
 			return j.RHS.PushPredicate(expr, semTable)
 		}
-		semTable.ProjectionErr = vterrors.Errorf(vtrpcpb.Code_UNIMPLEMENTED, "unsupported: cross-shard left join and where clause")
-		return nil
```
Comment (GuptaManan100) [inline go/vt/vtgate/planbuilder/abstract/join.go]: Why revert this change? Isn't it okay to keep it as a `ShardedErr`?

Comment (systay) [reply go/vt/vtgate/planbuilder/abstract/join.go]: Well, we have to either return an error, or build something that adds the predicate to the operator tree. At the moment we don't have a filter operator, so this will be easy to fix post-remove-query-trees
